pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -149,22 +171,80 @@ type CiliumBGPVirtualRouter struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Minimum=0
 	// +kubebuilder:validation:Maximum=4294967295
-	LocalASN int `json:"localASN"`
+	LocalASN int64 `json:"localASN"`
 	// ExportPodCIDR determines whether to export the Node's private CIDR block
 	// to the configured neighbors.
 	//
 	// +kubebuilder:validation:Optional
-	ExportPodCIDR bool `json:"exportPodCIDR"`
+	// +kubebuilder:default=false
+	ExportPodCIDR *bool `json:"exportPodCIDR,omitempty"`
 	// ServiceSelector selects a group of load balancer services which this
 	// virtual router will announce.
 	//
-	// If nil no services will be announced.
+	// If empty / nil no services will be announced.
 	//
 	// +kubebuilder:validation:Optional
-	ServiceSelector *slimv1.LabelSelector `json:"serviceSelector"`
+	ServiceSelector *slimv1.LabelSelector `json:"serviceSelector,omitempty"`
 	// Neighbors is a list of neighboring BGP peers for this virtual router
 	//
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinItems=1
 	Neighbors []CiliumBGPNeighbor `json:"neighbors"`
 }
+
+// SetDefaults applies default values on the CiliumBGPPeeringPolicy.
+// This is normally done by kube-apiserver for fields with explicit static defaults,
+// the main use of this method is to avoid the need for nil-checks in the controller code.
+func (p *CiliumBGPPeeringPolicy) SetDefaults() {
+	for i := range p.Spec.VirtualRouters {
+		p.Spec.VirtualRouters[i].SetDefaults()
+	}
+}
+
+// SetDefaults applies default values on the CiliumBGPVirtualRouter.
+// This is normally done by kube-apiserver for fields with explicit static defaults,
+// the main use of this method is to avoid the need for nil-checks in the controller code.
+func (r *CiliumBGPVirtualRouter) SetDefaults() {
+	if r.ExportPodCIDR == nil {
+		r.ExportPodCIDR = pointer.Bool(DefaultBGPExportPodCIDR)
+	}
+	for i := range r.Neighbors {
+		r.Neighbors[i].SetDefaults()
+	}
+}
+
+// SetDefaults applies default values on the CiliumBGPNeighbor.
```

Comment:
This is great, because it is possible (using `kubectl apply --force` or similar) to bypass the validation checks. Nice work.